lib/presentation/pages/movies_list.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:flutter/material.dart';
+
+import '../../interfaces/i_repository.dart';
+import '../../models/movies_model.dart';
+import '../../repositories/movie_repository.dart';
+import '../../ui_constants.dart';
+import '../widgets/movie_list_header.dart';
+import '../widgets/movie_list_item.dart';
+import '../widgets/plain_text.dart';
+
+class MoviesList extends StatefulWidget {
+  const MoviesList({
+    super.key,
+  });
+
+  static const String routeName = '/moviesList';
+
+  @override
+  State<MoviesList> createState() => _MoviesListState();
+}
+
+class _MoviesListState extends State<MoviesList> {
+  late Future<MoviesModel> futureMovies;
+  final IRepository repository = MovieRepository();
+
+  @override
+  void initState() {
+    futureMovies = repository.getModel() as Future<MoviesModel>;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return SafeArea(
+      child: Scaffold(
+        backgroundColor: UIConstants.backgroundColor,
+        body: FutureBuilder<MoviesModel>(
+            future: futureMovies,
```

Comment:
Here you can directly call `repository.getModel()`

Reply:
I advised her not to, according to [this documentation](https://api.flutter.dev/flutter/widgets/FutureBuilder-class.html?v=1.0.19&gclid=Cj0KCQjwl8anBhCFARIsAKbbpySmbqRANBZqcpoxps6sUPXj0nzUhn_NNXfMGq4ZmwF-fRJCFiema_AaAgLTEALw_wcB&gclsrc=aw.ds#:~:text=a%20Future.-,Managing%20the%20future,-The%20future%20must). Thoughts @FranciscoGaraventa ?